onnxruntime/core/mlas/lib/mlasi.h

```diff
@@ -311,6 +317,24 @@ operator!=(const MLFloat16& left, const MLFloat16& right)
 
 #endif  // BUILD_MLAS_NO_ONNXRUNTIME
 
+#if defined(MLAS_TARGET_ARM64)
+
+struct SMEInfo {
+    static const bool CanUseSME2;
+    static const bool CanUseSME;
+    static const bool IsSMEAvailable;
+};
+
+// Boolean condition to determine if we can use SME2
+// By default we should try for SME2 first before falling back to SME.
+inline const bool SMEInfo::CanUseSME2 = MLAS_CPUIDINFO::GetCPUIDInfo().HasArm_SME2();
```

Comment:
`CPUIDInfo::HasArm_SME2()` already returns a cached bool. is it worth having another constant for the same value? admittedly, `SMEInfo::CanUseSME2` is less to type, but it may be simpler to just have a single way of getting this information.